### PR TITLE
feat(avo-2599): add share flyout on assignment page

### PR DESCRIPTION
**Pull request by SHDBramVandenbussche**
[AVO-2599](https://meemoo.atlassian.net/browse/AVO-2599)

[AVO2-Proxy PR](https://github.com/viaacode/avo2-proxy/pull/621)
[AVO2-Components PR](https://github.com/viaacode/avo2-components/pull/527)
[AVO2-Typings PR](https://github.com/viaacode/avo2-typings/pull/115)

Before:
image

After:
image

[AVO-2599]: https://meemoo.atlassian.net/browse/AVO-2599?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by WardVerc1 — APPROVED** [review]

**Review by WardVerc1 — APPROVED** [review]

File: src/assignment/components/AssignmentActions.tsx [hunk 1]
```diff
@@ -33,9 +41,95 @@ const AssignmentActions: FunctionComponent<AssignmentActionsProps> = ({
...
+		const response = await AssignmentService.fetchContributorsByAssignmentId(
+			share?.assignment?.id
+		);
+
+		setContributors(response as Contributor[]);
+	}, [share]);
+
+	useEffect(() => {
+		fetchContributors();
+	}, [share]);
```
Comment (bertyhell) [inline src/assignment/components/AssignmentActions.tsx:63]: you don't need to repeat the [share] dependency from the useCallback function here. You can just make the useCallback function the dependency:
```suggestion
	}, [fetchContributors]);
```

File: src/shared/components/ShareDropdown/ShareDropdown.tsx [hunk 2]
```diff
@@ -42,16 +46,24 @@ const ShareDropdown: FC<ShareDropdownProps> = ({
...
+		if (!hasModalOpen) {
+			setIsShareDropdownOpen(false);
+		}
+	};
+
 	return (
 		<Dropdown
 			isOpen={isShareDropdownOpen}
-			onClose={() => setIsShareDropdownOpen(false)}
+			onClose={() => handleOnClose()}
```
Comment (bertyhell) [inline src/shared/components/ShareDropdown/ShareDropdown.tsx:64]:
```suggestion
			onClose={handleOnClose}
```

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx [hunk 3]
```diff
@@ -108,36 +147,50 @@ const ShareWithColleagues: FC<ShareWithColleaguesProps & UserProps> = ({
...
+													? (
+															contributor.firstName[0] +
+															contributor.lastName[0]
+													  ).toLocaleUpperCase()
+													: contributor.email?.slice(0, 2).toUpperCase()
+											}
+											image={contributor.profileImage}
+										/>
+									) : (
+										<div className="c-colleague-info-row__avatar--pending" />
```
Comment (bertyhell) [inline src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx:163]: empty row for pending? or is the email address for the invited user added later?

File: src/shared/helpers/transform-contributors.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import {
+	ContributorInfo,
+	ContributorInfoRights,
+} from '../components/ShareWithColleagues/ShareWithColleagues.types';
+import { Contributor } from '../types/contributor';
+
+export const transformContributorsToSimpleContributors = (
+	owner: Avo.User.User,
+	contributors: Contributor[]
+) => {
```
Comment (bertyhell) [inline src/shared/helpers/transform-contributors.ts:12]:
```suggestion
): ContributorInfo[] => {
```

File: src/shared/helpers/transform-contributors.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	if (!owner || !contributors) {
+		return;
```
Comment (bertyhell) [inline src/shared/helpers/transform-contributors.ts:14]:
```suggestion
		return [];
```

**Review by bertyhell — APPROVED** [review]

File: src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx (hunk 3 above)
Comment (SHDBramVandenbussche) [reply src/shared/components/ShareWithColleagues/ShareWithColleagues.tsx:163]: The empty div is a circle with a dashed border to replace the non existing avatar

File: src/assignment/assignment.service.ts [hunk 6]
```diff
@@ -1525,4 +1533,89 @@ export class AssignmentService {
 			});
 		}
 	}
+
+	static async fetchContributorsByAssignmentId(assignmentId: string) {
```
Comment (bertyhell) [inline src/assignment/assignment.service.ts:1537]: provide return types for all service methods
```
	static async fetchContributorsByAssignmentId(assignmentId: string): Promise<...> {
```

**Review by bertyhell — APPROVED** [review]

File: src/assignment/assignment.service.ts [hunk 7]
```diff
@@ -1566,7 +1567,10 @@ export class AssignmentService {
 		}
 	}
 
-	static async addShareAssignmentUser(assignmentId: string, user: Partial<ContributorInfo>) {
+	static async addShareAssignmentUser(
+		assignmentId: string,
+		user: Partial<ContributorInfo>
+	): Promise<void> {
 		try {
 			return await fetchWithLogoutJson(
```
Comment (bertyhell) [inline src/assignment/assignment.service.ts:1575]:
```suggestion
			await fetchWithLogoutJson(
```

File: src/assignment/assignment.service.ts [hunk 8]
```diff
@@ -1603,7 +1607,10 @@ export class AssignmentService {
 		}
 	}
 
-	static async deleteShareAssignmentUser(assignmentId: string, contributorId: string) {
+	static async deleteShareAssignmentUser(
+		assignmentId: string,
+		contributorId: string
+	): Promise<void> {
 		try {
 			return await fetchWithLogoutJson(
```
Comment (bertyhell) [inline src/assignment/assignment.service.ts:1615]:
```suggestion
			await fetchWithLogoutJson(
```

**Review by bertyhell — APPROVED** [review]

**Review by bertyhell — APPROVED** [review]

**Pull request merged**